### DISCUSSION TITLE
All input videos may not be mp4

### DISCUSSION THREAD

**Discussion by technosaby** [technosaby/gsoc2022#21] [General]
- All video in database may not be mp4
- Also mp4 videos may also be different  based on how they are created (like ipTV from Macintosh may have different format)

So may need to check 
- if my parser works for all mp4 videos, If not, check why ?
- Also support for other formats ?

**Comment by brucearctor** [comment]
ideal if you can parse different types of video.

Though, I think there is an existing audio-extraction process for another pipeline...   We might want to understand that.  And, determine whether can use that to get the audio.  TBD:  And whether sufficiently modular - as unclear whether worth refactoring the existing code, vs writing fresh.  

Do check with Mark and others, I hope they can answer the question of whether Red Hen videos are standardized.  Though, even better if what you develop can be useful for things beyond existing red hen data.

**Comment by technosaby** [comment]
@brucearctor I think my parser, which used ffmpeg, will parse most container formats natively, including MP4, . ts, MOV, AVI, Y4M, MKV, and many others. So I think I will be good for this project. Do you agree ?

**Comment by brucearctor** [comment]
ffmpeg is very common and versatile, so a solid choice/foundation. 

For now, most important thing is will work with Red Hen things -- and being able to work with additional formats is a nice bonus!
